java/fury-core/src/main/java/org/apache/fury/serializer/ArraySerializers.java

```diff
@@ -85,52 +103,69 @@ public void write(MemoryBuffer buffer, T[] arr) {
       int len = arr.length;
       RefResolver refResolver = fury.getRefResolver();
       Serializer componentSerializer = this.componentTypeSerializer;
-      int header = componentSerializer != null ? 0b1 : 0b0;
-      buffer.writeVarUint32Small7(len << 1 | header);
-      if (componentSerializer != null) {
-        for (T t : arr) {
-          if (!refResolver.writeRefOrNull(buffer, t)) {
-            componentSerializer.write(buffer, t);
-          }
-        }
+      if (this.collectionSerializer != null) {
+        fury.getGenerics().pushGenericType(this.collectionGenericType);
+        ArrayAsList list = new ArrayAsList(0);
+        list.setArray(arr);
+        this.collectionSerializer.write(buffer, list);
```

Comment:
We need to invoke `fury.incDepth` before  and after invoking `collectionSerializer.write`

Reply:
I tried with 

incDepth(1)
...
incDepth(1)

Should I be using -1 after serialization to reset the depth?

Reply:
incDepth(-1) should be invoked after serialization